lib/src/document_reader.dart

```diff
@@ -93,11 +95,22 @@ class DocumentReader<DocType extends DocumentData> extends StateNotifier<Documen
     _setState(DocumentReaderConnecting());
     await nfc.connect(iosAlertMessage: iosNfcMessages(DocumentReaderConnecting()));
 
-    _AuthMethod method = _AuthMethod.bac;
-    _setState(DocumentReaderAuthenticating());
-    final bacSuccess = await tryAuthenticateWithBAC();
+    _AuthMethod method;
+    bool bacSuccess = false;
+    if (useBAC == false) {
+      method = _AuthMethod.pace;
+    } else {
+      method = _AuthMethod.bac;
+      _setState(DocumentReaderAuthenticating());
+      bacSuccess = await tryAuthenticateWithBAC();
+    }
+
+    // failure when bac is not successful?
+    if (!bacSuccess && useBAC == null) {
```

Comment:
Why not nest this condition inside the `else` branch above?
And, why not reverse the order of the branches, so it becomes easier to read?